Features/Context/PlatformUI.php

```diff
@@ -0,0 +1,423 @@
+<?php
+
+/**
+ * File containing the main context class for PlatformUI
+ * This flie contains the mapping of the BDD sentences to the functions that implement them
+ * Also contains the necessary initializations
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\PlatformUIBundle\Features\Context;
+
+use EzSystems\BehatBundle\Context\Browser\Context;
+use EzSystems\PlatformUIBundle\Features\Helper\JavaScript as JsHelper;
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit_Framework_Assert as Assertion;
+
+class PlatformUI extends Context
+{
+    use SubContext\Authentication;
+    use SubContext\CommonActions;
+
+    /**
+     * PlatformUI relative URL path
+     *
+     * @var string
+     */
+    private $platformUiUri;
+
+    /**
+     * WebDriver in use
+     * Only Sahi or Selenium2 accepted
+     *
+     * @var string
+     */
+    protected $driver;
+
+    /**
+     * Evaluate javascript code and return result.
+     *
+     * @param   string  $jsCode
+     * @return  mixed
+     */
+    protected function evalJavascript( $jsCode, $addJsHelper = true )
+    {
+        if ( $addJsHelper )
+        {
+            $jsCode = JsHelper::getHelperJs() . $jsCode;
+        }
+        $jsCode = $this->driver->wrappedFunction( $jsCode );
+
+        return $this->getSession()->evaluateScript( $jsCode );
+    }
+
+    /**
+     * Execute javascript code and return result.
+     *
+     * @param   string  $jsCode
+     * @return  mixed
+     */
+    protected function execJavascript( $jsCode, $addJsHelper = true )
+    {
+        if ( $addJsHelper )
+        {
+            $jsCode = JsHelper::getHelperJs() . $jsCode;
+        }
+
+        $jsCode = $this->driver->wrappedFunction( $jsCode );
+
+        return $this->getSession()->executeScript( $jsCode );
+    }
+
+    /**
+     * Waits for Javascript to finnish by running a empty Javascript
+     * (In Sahi it's possible to have the same result by running an empty javascript only)
+     */
+    protected function waitForLastJs()
+    {
+        //Needs to be here
```

Comment:
this is very confusing and instead we should be able to read the *why* it needs to be here.